### PR TITLE
Move to Cobra from go flags

### PR DESCRIPTION
**Pull request by alexellis**
Signed-off-by: Alex Ellis (OpenFaaS Ltd) <alexellis2@gmail.com>

## Description

Moving to Cobra allows for additional commands to be added
easily.

## How Has This Been Tested?

Tested e2e with k3d

```sh
/usr/local/bin/k3d create --server-arg="--no-deploy=traefik"
export KUBECONFIG="$(/usr/local/bin/k3d get-kubeconfig --name='k3s-default')"

go build && ./ofc-bootstrap apply --file example.init.yaml
```

Whilst using K8s 1.16 I fixed an issue with of-builder.
import-secrets is now patched via the API instead of `kubectl patch`

For the old syntax, users get a hint on what to use instead:

```sh
./ofc-bootstrap --yaml example.init.yaml
Error: a breaking change was introduced, you now need to use ofc-bootstrap apply --file init.yaml
```

## Checklist:

I have:

- [x] checked my changes follow the style of the existing code / OpenFaaS repos
- [ ] updated the documentation and/or roadmap in README.md
- [ ] read the [CONTRIBUTION](https://github.com/openfaas/faas/blob/master/CONTRIBUTING.md) guide
- [x] signed-off my commits with `git commit -s`
- [x] added unit tests

### CODE REVIEW

**Comment by alexellis** [comment]
cc @csakshaug

**Comment by Waterdrips** [comment]
CI Scripts need updating to new command syntax

**Comment by Waterdrips** [comment]
I will test this tomorrow locally

File: cmd/root.go [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+	SilenceUsage: true,
+}
+
+var versionCmd = &cobra.Command{
+	Use:   "version",
+	Short: "Display version information.",
+	Run:   parseBaseCommand,
+}
+
+func getVersion() string {
```
Comment (Waterdrips) [inline cmd/root.go]: Also my IDE is saying this is not used too

File: cmd/root.go [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+	// GitCommit as per git repo
+	GitCommit string
+)
+
+// WelcomeMessage to introduce ofc-bootstrap
+const WelcomeMessage = "Welcome to ofc-bootstrap! Find out more at https://github.com/openfaas-incubator/ofc-bootstrap"
+
+func init() {
+	rootCommand.AddCommand(versionCmd)
+	rootCommand.Flags().StringArrayP("yaml", "f", []string{""}, "The init.yaml plan file")
```
Comment (Waterdrips) [inline cmd/root.go]: is there a way to exclude this from the "usage" output? Its showing as a command and I though it was deprecated?

Screenshot 2019-12-04 at 07 59 12

Comment (alexellis) [reply cmd/root.go]: Which usage do you want to exclude? How did you get it to show?

**Comment by alexellis** [comment]
Thanks for letting me know about the unused methods, I am not worried about merging them at present, they can be removed easily later.

How was the functionality? Does TLS / OAuth still work as a setting?

File: cmd/root.go (hunk 2 above)
Comment (Waterdrips) [reply cmd/root.go]: running ./ofc-bootstrap creates the above screenshot output, --yaml has been depricated for --file, or -f?

Comment (alexellis) [reply cmd/root.go]: For `ofc-bootstrap` on its own I think that is OK

`ofc-bootstrap --yaml file.yaml` -> produces the warning.

**Comment by Waterdrips** [comment]
I have tested:
No TLS, TLS, no Oauth, Oauth
I got no errors and the cert requests and endpoints were configured correctly.

this was done in k3d on my laptop so no real certs were generated, but the requests were fired etc.
 
This looks good. LGTM

**Comment by alexellis** [comment]
Thank you

**Pull request merged**